_posts/news/2021-01-12-announcing-scalajs-1.4.0.md

```diff
@@ -0,0 +1,188 @@
+---
+layout: post
+title: Announcing Scala.js 1.4.0
+category: news
+tags: [releases]
+permalink: /news/2021/01/12/announcing-scalajs-1.4.0/
+---
+
+
+We are excited to announce the release of Scala.js 1.4.0!
+
+This release complements the module splitting support, introduced in Scala.js 1.3.0, with support for dynamic module loading.
+It is now possible to structure an application such that specific features are only loaded if necessary, the first time they are used.
+
+In addition, this release significantly improves the performance of `scala.Array`.
+In particular, arrays of numeric types (except Longs) are implemented with JavaScript typed arrays under the hood.
+
+This release also contains a number of bug fixes.
+The version of the Scala standard library for 2.13.x was upgraded to 2.13.4.
+
+Read on for more details.
+
+<!--more-->
+
+## Getting started
+
+If you are new to Scala.js, head over to [the tutorial]({{ BASE_PATH }}/tutorial/).
+
+If you need help with anything related to Scala.js, you may find our community [on Gitter](https://gitter.im/scala-js/scala-js) and [on Stack Overflow](https://stackoverflow.com/questions/tagged/scala.js).
+
+Bug reports can be filed [on GitHub](https://github.com/scala-js/scala-js/issues).
+
+## Release notes
+
+If upgrading from Scala.js 0.6.x, make sure to read [the release notes of Scala.js 1.0.0]({{ BASE_PATH }}/news/2020/02/25/announcing-scalajs-1.0.0/) first, as they contain a host of important information, including breaking changes.
+
+This is a **minor** release:
+
+* It is backward binary compatible with all earlier versions in the 1.x series: libraries compiled with 1.0.x through 1.3.x can be used with 1.4.0 without change.
+* It is *not* forward binary compatible with 1.3.x: libraries compiled with 1.4.0 cannot be used with 1.3.x or earlier.
+* It is *not* entirely backward source compatible: it is not guaranteed that a codebase will compile *as is* when upgrading from 1.3.x (in particular in the presence of `-Xfatal-warnings`).
+
+As a reminder, libraries compiled with 0.6.x cannot be used with Scala.js 1.x; they must be republished with 1.x first.
+
+## Fixes with compatibility concerns
```

Comment:
I wonder if the following need mentioning (probably not, just want to point it out):

https://github.com/scala-js/scala-js/commit/0b51d7bdc93a5df738f4c09aabf03bbb6f862685
https://github.com/scala-js/scala-js/issues/4212

Reply:
No, I don't think so. I've added https://github.com/scala-js/scala-js/issues/4212 to the list of issues at the end, though.